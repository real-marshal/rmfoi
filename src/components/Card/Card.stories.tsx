import type { Meta, Story } from '@storybook/react'
import { Card, CardProps } from './Card'
import BlankImage from '@/assets/blank.png'

export default {
  title: 'Common/Card',
  component: Card,
} as Meta

const Template: Story<CardProps> = (args) => <Card {...args} />

export const Basic = Template.bind({})
Basic.args = {
  name: 'Card heading',
  description: 'This is the description of the content that this card is supposed to encapsulate',
  media: 'image',
  src: BlankImage,
}
