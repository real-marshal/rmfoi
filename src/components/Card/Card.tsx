import { dt } from '@/features/Theming'
import styled from '@emotion/styled'
import { Link } from 'react-router-dom'
import { Heading } from '../Heading'

interface CardProps {
  name: string
  description: string
  media: 'image' | 'video'
  src: string
  link?: string
}

const Container = styled.div`
  display: flex;
  flex-direction: column;
  height: fit-content;
  width: 100%;
  color: ${dt.theme.primary};
  background-color: ${dt.theme.background};
  border: 3px solid ${dt.theme.primary};
  border-radius: 10px;

  & > :not(img, a) {
    padding: ${dt.padding.lg};
  }
`

const Image = styled.img`
  width: 100%;
  border-radius: 10px;
`

const Video = styled.video`
  width: 100%;
  border-radius: 10px;
`

const Description = styled.p`
  margin: 0;
  font-size: 1.2rem;
`

const StyledHeading = styled(Heading)`
  padding: ${dt.padding.lg};
  background: ${dt.theme.gradientGreenBlue};
  border-radius: 10px 10px 0 0;
  border-bottom: 3px solid ${dt.theme.primary};
  font-size: 1.25rem;
`

const Card = ({ name, description, media, src, link }: CardProps) => {
  const heading = (
    <StyledHeading type='h3' underline={false}>
      {name}
    </StyledHeading>
  )
  return (
    <Container>
      {link ? <Link to={link}>{heading}</Link> : heading}
      <Description>{description}</Description>
      {media === 'image' && <Image src={src} alt={`${name} overview`} />}
      {media === 'video' && <Video src={src} autoPlay loop muted playsInline />}
    </Container>
  )
}

export { Card }
export type { CardProps }
