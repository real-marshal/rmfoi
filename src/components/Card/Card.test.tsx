import { composeStories } from '@storybook/react'
import { render } from '@testing-library/react'
import * as stories from './Card.stories'

describe('Card component', () => {
  const testStories = Object.values(composeStories(stories)).map(
    (Story) => [Story.storyName, Story] as [string, typeof Story]
  )

  it.each(testStories)('renders %s story', (_, Story) => {
    const view = render(<Story />)
    expect(view.baseElement).toMatchSnapshot()
  })
})
