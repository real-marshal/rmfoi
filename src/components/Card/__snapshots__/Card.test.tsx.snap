// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders Basic story 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  width: 100%;
  color: var(--dt-theme-primary);
  background-color: var(--dt-theme-background);
  border: 3px solid var(--dt-theme-primary);
  border-radius: 10px;
}

.emotion-0>:not(img, a) {
  padding: var(--dt-padding-lg);
}

.emotion-3 {
  font-family: monospace;
  font-weight: regular;
  color: var(--dt-theme-primary);
  margin: 0;
  padding: var(--dt-padding-lg);
  background: var(--dt-theme-gradientGreenBlue);
  border-radius: 10px 10px 0 0;
  border-bottom: 3px solid var(--dt-theme-primary);
  font-size: 1.25rem;
}

.emotion-5 {
  margin: 0;
  font-size: 1.2rem;
}

.emotion-7 {
  width: 100%;
  border-radius: 10px;
}

<body>
  <div>
    <div
      class="emotion-0 emotion-1"
    >
      <h3
        class="emotion-2 emotion-3 emotion-4"
      >
        Card heading
      </h3>
      <p
        class="emotion-5 emotion-6"
      >
        This is the description of the content that this card is supposed to encapsulate
      </p>
      <img
        alt="Card heading overview"
        class="emotion-7 emotion-8"
        src="file-mock"
      />
    </div>
  </div>
</body>
`;
